src/Microsoft.AspNetCore.Mvc.Core/Resources.resx

```diff
@@ -410,7 +410,7 @@
     <value>No page named '{0}' matches the supplied values.</value>
   </data>
   <data name="UrlHelper_RelativePagePathIsNotSupported" xml:space="preserve">
-    <value>The relative page path '{0}' can only be used while executing a Razor Page. Specify a root relative path with a leading '/' to generate a URL outside of a Razor Page.</value>
+    <value>The relative page path '{0}' can only be used while executing a Razor Page. Specify a root relative path with a leading '/' to generate a URL outside of a Razor Page. If you are using {1} then you must provide the current {2} to use relative pages.</value>
```

Comment:
You can hit this case when using `GetPathByPage(..)` without an HttpContext. This needed to be clarified because otherwise it would be really confusing.